### ISSUE TITLE
Asking for newbie tutorial.

### ISSUE THREAD

**Issue by meyverick** [nodelord/facepunch_rust_exporter#1]
Hello,
This looks very interesting but I'm newbie and I don't understand how to install and use it.
Would you like to help me? I downloaded the Windows release and ran v2.exe... it only opens a cmd windows.

**Comment by jamesalbert** [comment]
@meyverick

This is a really good question. Full transparency, we're working on providing some more documentation on setup and configuration, + renaming the release executable to a more friendly name 😄 

but for right now, I'll post some info that should hopefully help you out:

First, this exporter plugs into a telemetry stack. Minimally, this is what it looks like:

- victoriametrics (where your metrics will be stored)
- exporter <- you are here (an http server that listens for requests. When invoked, it scrapes metrics from the rust server and sends them back to the caller in the form of Prometheus formatted metrics)
- vmagent (software that is configured to invoke your exporter at a specified interval and send the data off to victoriametrics)
- grafana (graphs and stuff)

When you run the release binary, you are setting up a local server. It doesn't show any graphs or anything. You'll have to configure and run victoriametrics, vmagent, and grafana alongside it, or else it does nothing.

It's a bit to wrap your head around at first, and until we upload some better docs, I would definitely suggest checking out videos on youtube searching "prometheus grafana docker" (this one looks cool: https://www.youtube.com/watch?v=83LWo7h_hvs). You'll probably find on there a lot of people using/setting up `node_exporter` (an exporter for general systems metrics), but all these exporters do the same thing, they just target different platforms/software/etc.


here's a diagram of a general prometheus workflow:
![image](https://user-images.githubusercontent.com/1617698/169906265-07c7c112-e151-4e5e-b92c-7b1c0afed20b.png)

In short,

- vmagent makes a request to exporter
- exporter scrapes data from source (in this case, calls rust rcon commands on the target server) and sends it back to vmagent
- vmagent takes this data and sends it off to victoriametrics
- grafana pulls data from victoriametrics and displays it in graphs

edit: clarification

**Comment by meyverick** [comment]
Thank you for your explanation, it will help me a lot. I'll try to get a result and wait for your docs if I don't manage to get it.
